driver-core/src/test/java/com/datastax/driver/core/querybuilder/QueryBuilderUDTExecutionTest.java

```diff
@@ -87,4 +88,39 @@ public void should_handle_collections_of_UDT() throws Exception {
         r1 = rows.get(0);
         assertThat(r1.getMap("m", Integer.class, UDTValue.class)).isEqualTo(map);
     }
+
+    /**
+     * Ensures that UDT fields can be set and retrieved on their own using {@link QueryBuilder#set} and
+     * {@link QueryBuilder#select} respectively.
+     *
+     * @test_category queries:builder
+     * @jira_ticket JAVA-1286
+     * @jira_ticket CASSANDRA-7423
+     */
+    @CassandraVersion(major=3.6, description="Requires CASSANDRA-7423 introduced in Cassandra 3.6")
+    @Test(groups = "short")
+    public void should_support_setting_and_retrieving_udt_fields() {
+        //given
+        String table = "unfrozen_udt_table";
+        String udt = "person";
+        session().execute(createType(udt).addColumn("first", DataType.text()).addColumn("last", DataType.text()));
+        UserType userType = cluster().getMetadata().getKeyspace(keyspace).getUserType(udt);
+        assertThat(userType).isNotNull();
+
+        session().execute(createTable(table).addPartitionKey("k", DataType.text())
+                .addUDTColumn("u", udtLiteral(udt))
```

Comment:
The name refers to expressing the type with a string literal, I think this still holds even if the type is unfrozen.